src/main/java/com/spring/boot/repository/CommentRepository.java

```diff
@@ -1,9 +1,19 @@
 package com.spring.boot.repository;
 
 import com.spring.boot.model.Comment;
+import java.util.List;
+import java.util.Optional;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.stereotype.Repository;
 
 @Repository
 public interface CommentRepository extends JpaRepository<Comment, Long> {
+    @Query(value = "SELECT c FROM Comment c "
+            + "WHERE c.productId = ?1")
+    List<Optional<Comment>> getProductComments(String productId);
```

Comment:
I don't see the point of fetching Optionals. (you don't check them anyway) Se what you will get if there is no result (empty list or exception)